src/main/java/com/epam/task/exceptions/PurchaseList.java

```diff
@@ -0,0 +1,104 @@
+package com.epam.task.exceptions;
+
+import com.epam.task.exceptions.customExc.CsvLineException;
+
+import java.io.*;
+import java.util.*;
+
+public class PurchaseList{
+    private final ArrayList<AbstractPurchase> purchaseList;
+    private final Comparator<AbstractPurchase> comparator;
+    private boolean sorted = false;
+
+    public PurchaseList(String filePath, Comparator<AbstractPurchase> comparator){
+        this(new File(filePath), comparator);
+    }
+
+    public PurchaseList(Comparator<AbstractPurchase> comparator){
+        this.purchaseList = new ArrayList<>(0);
+        this.comparator = comparator;
+    }
+
+    public PurchaseList(File inFile, Comparator<AbstractPurchase> comparator) {
+        this(comparator);
+
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(inFile));
+            String line;
+            while ((line = bufferedReader.readLine()) != null){
+                PurchaseFactory purchaseFactory = new PurchaseFactory();
+                AbstractPurchase purchase = null;
+                try {
+                    purchase = purchaseFactory.createPurchase(line);
+                } catch (CsvLineException e) {
+                    System.err.println(e + " caused by: " + e.getCause().getClass().getSimpleName() + ": "
+                            + e.getCause().getMessage());
+                }
+
+                if(purchase != null)
+                    this.purchaseList.add(purchase);
+
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void insertPurchase(AbstractPurchase purchase, int ind){
+        try{
+            this.purchaseList.add(ind, purchase);
+        }catch (Exception e){
+            this.purchaseList.add(purchase);
+        }
```

Comment:
J. Bloch, 3rd edition. 
Item 69: Use exceptions only for exceptional conditions